### PR TITLE
Feature/fix responsive overrides #17

### LINKED ISSUE #17: Responsive styles get overridden
Given this css:

```css
.foo {
    background: url(some_image.png) 0 0 no-repeat at-2x;
}

@media (min-width: 768px) {
    .foo {
        background-position: -6px;
    }
}
```

postcss-at2x will output:

```css
.foo {
    background: url(some_image.png) 0 0 no-repeat;
}

@media (min-width: 768px) {
    .foo {
        background-position: -6px;
    }
}

@media not all, (min-resolution: 144dpi), (min-resolution: 1.5dppx) {
    .foo {
        background: url(some_image@2x.png) 0 0 no-repeat;
        background-size: 36px 36px;
    }
}
```

The `resolution` media query is after the `min-width` query, so `background-position` gets overridden.

I see two potential fixes:
1. Put the resolution query immediately after the initial declaration blocks, so order specificity is maintained
2. Only set the `background-image`

2 looks like the easiest option, I'm going to take a look at that now.

### PR DESCRIPTION
**Pull request by tdhooper**
Fixes #17

### CODE REVIEW

**Comment by simonsmith** [comment]
@tdhooper Sorry for my delay here. Will review this today or tomorrow. Thanks for your help again!

File: lib/index.js [hunk 1]
```diff
@@ -159,28 +151,58 @@ function combineMediaQuery(queries, resolutions) {
...
-      break;
-    }
+// Returns the <image> part of the background definition,
+// and includes the identifier if it meets the criteria:
+// * It's a url() image
+// * It's not an svg
+// * The background definition has at-2x applied
+function extractRetinaImage(identifier, background) {
+  const match = background.match(imageRegex);
+  const [image, type, value] = match;
```
Comment (simonsmith) [inline lib/index.js]: Do we need to check if match could be `null` here?

Comment (tdhooper) [inline lib/index.js]: Good spot!

**Comment by simonsmith** [comment]
Great stuff. Cheers

File: lib/index.js [hunk 2]
```diff
@@ -159,28 +151,63 @@ function combineMediaQuery(queries, resolutions) {
...
+      backgrounds[backgrounds.length - 1] += `,${part}`;
+    } else {
+      backgrounds.push(part);
+    }
+    const open = count(part, '(');
+    const close = count(part, ')');
+    opened += open - close;
+  });
+  return backgrounds;
+}
```
Comment (simonsmith) [inline lib/index.js]: @tdhooper Curious about the intention of this function. Could the same not be achieved with `return value.split(',')`?

**Comment by tdhooper** [comment]
That was my initial approach, but it returns separate parts for the linear-gradient syntax.

Then again, the rest of the code should cope fine with that now.

**Comment by tdhooper** [comment]
Looking at this a bit more, for the given css:

```css
.multi {
  background: url(http://example.com/image.png),
                    linear-gradient(to right, rgba(255, 255, 255, 0),  rgba(255, 255, 255, 1)),
                    green,
                    url(/public/images/cool.png) at-2x;
}
```

I don't think we should repeat `green` in the `background-image` rule we create, as it's not a valid image.

Instead, `none` makes more sense:

```css
.multi {
  background-image: url(http://example.com/image.png),
                    linear-gradient(to right, rgba(255, 255, 255, 0),  rgba(255, 255, 255, 1)),
                    none,
                    url(/public/images/cool@2x.png);
}
```

**Pull request merged**